pkg/routeagent/controllers/route/route.go

```diff
@@ -681,11 +688,10 @@ func (r *Controller) processNextEndpoint() bool {
 		}
 		// If the active Gateway transitions to a new node, we flush the HostNetwork routing table.
 		r.updateRoutingRulesForHostNetworkSupport(nil, FlushRouteTable)
-		r.gwVxLanMutex.Unlock()
 
-		// NOTE(mangelajo): This may not belong here, it's a gateway cleanup thing
-		r.cleanStrongswanRoutingTable()
-		r.cleanXfrmPolicies()
+		r.GatewayToNonGatewayTransition()
```

Comment:
The order of the mutex has been modified after talking with @sridhargaddam as it seemed the safest